controllers/component_image_controller.go

```diff
@@ -331,32 +347,114 @@ func (r *ComponentReconciler) reportError(ctx context.Context, component *appstu
 	log := ctrllog.FromContext(ctx)
 
 	robotAccountSecret := generateSecret(component, robotAccount, secretName, imageURL)
+	secretData := robotAccountSecret.StringData
+
 	robotAccountSecretKey := types.NamespacedName{Namespace: robotAccountSecret.Namespace, Name: robotAccountSecret.Name}
 	existingRobotAccountSecret := &corev1.Secret{}
 	if err := r.Client.Get(ctx, robotAccountSecretKey, existingRobotAccountSecret); err == nil {
-		existingRobotAccountSecret.StringData = robotAccountSecret.StringData
+		existingRobotAccountSecret.StringData = secretData
 		if err := r.Client.Update(ctx, existingRobotAccountSecret); err != nil {
 			log.Error(err, fmt.Sprintf("failed to update robot account secret %v", robotAccountSecretKey), l.Action, l.ActionUpdate)
-			return err
+			return nil, err
 		}
 	} else {
 		if !errors.IsNotFound(err) {
 			log.Error(err, fmt.Sprintf("failed to read robot account secret %v", robotAccountSecretKey), l.Action, l.ActionView)
-			return err
+			return nil, err
 		}
-		if err := r.Client.Create(ctx, &robotAccountSecret); err != nil {
+		if err := r.Client.Create(ctx, robotAccountSecret); err != nil {
 			log.Error(err, fmt.Sprintf("error writing robot account token into Secret: %v", robotAccountSecretKey), l.Action, l.ActionAdd)
+			return nil, err
+		}
+	}
+
+	return secretData, nil
+}
+
+// EnsureComponentPullSecretRemoteSecret creates remote secret for component image repository pull token.
+func (r *ComponentReconciler) EnsureComponentPullSecretRemoteSecret(ctx context.Context, component *appstudioredhatcomv1alpha1.Component, remoteSecretName string) error {
+	log := ctrllog.FromContext(ctx)
+
+	remoteSecret := &remotesecretv1beta1.RemoteSecret{}
+	remoteSecretKey := types.NamespacedName{Namespace: component.Namespace, Name: remoteSecretName}
+	if err := r.Client.Get(ctx, remoteSecretKey, remoteSecret); err != nil {
+		if !errors.IsNotFound(err) {
+			log.Error(err, fmt.Sprintf("failed to get remote secret: %v", remoteSecretKey), l.Action, l.ActionAdd)
+			return err
+		}
+
+		remoteSecret := &remotesecretv1beta1.RemoteSecret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      remoteSecretName,
+				Namespace: component.Namespace,
+				Labels: map[string]string{
+					ApplicationNameLabelName: component.Spec.Application,
+					ComponentNameLabelName:   component.Name,
+				},
+				OwnerReferences: []metav1.OwnerReference{
+					{
+						Name:       component.Name,
+						APIVersion: component.APIVersion,
+						Kind:       component.Kind,
+						UID:        component.UID,
+					},
+				},
+			},
+			Spec: remotesecretv1beta1.RemoteSecretSpec{
+				Secret: remotesecretv1beta1.LinkableSecretSpec{
+					Name: remoteSecretName,
+					Type: corev1.SecretTypeDockerConfigJson,
+					LinkedTo: []remotesecretv1beta1.SecretLink{
+						{
+							ServiceAccount: remotesecretv1beta1.ServiceAccountLink{
+								Reference: corev1.LocalObjectReference{
+									Name: BuildPipelineServiceAccountName,
```

Comment:
This pull secret should not be linked to appstudio-pipeline service account, since there is already linked push secret which is using same repo path.

Reply:
Currently user workload is running under `default` SA.

Reply:
Thanks, fixed.